ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java

```diff
@@ -110,15 +185,130 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
+                    break;
+            }
+        }
+        ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.currentToken(), parser);
+    }
+
+    private static void parseBucket(XContentParser parser, Output.Bucket bucket) throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            switch (parser.currentName()) {
+                case Output.Bucket.FIELD_START_TIME:
+                    parser.nextToken();
+                    bucket.setStartTime(parser.longValue());
+                    break;
+                case Output.Bucket.FIELD_END_TIME:
+                    parser.nextToken();
+                    bucket.setEndTime(parser.longValue());
+                    break;
+                case Output.Bucket.FIELD_OVERALL_VALUE:
+                    parser.nextToken();
+                    bucket.setOverallAggValue(parser.doubleValue());
+                    break;
+                case Output.Bucket.FIELD_ENTITIES:
+                    parseEntities(parser, bucket);
+                    break;
+            }
+        }
+        ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.currentToken(), parser);
+    }
+
+    private static void parseEntities(XContentParser parser, Output.Bucket bucket) throws IOException {
+        List<Output.Entity> entities = new ArrayList<>();
+        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+            Output.Entity entity = new Output.Entity();
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                switch (parser.currentName()) {
+                    case Output.Entity.FIELD_KEY:
+                        List<String> key = new ArrayList<>();
+                        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser);
+                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                            key.add(parser.text());
+                        }
+                        entity.setKey(key);
+                        break;
+                    case Output.Entity.FIELD_CONTRIBUTION_VALUE:
+                        parser.nextToken();
+                        entity.setContributionValue(parser.doubleValue());
+                        break;
+                    case Output.Entity.FIELD_BASE_VALUE:
+                        parser.nextToken();
+                        entity.setBaseValue(parser.doubleValue());
+                        break;
+                    case Output.Entity.FIELD_NEW_VALUE:
+                        parser.nextToken();
+                        entity.setNewValue(parser.doubleValue());
+                        break;
```

Comment:
Skip children for default case? Same for other places
```
                default:
                    parser.skipChildren();
                    break;
```

Reply:
done